tools/ts_library_builder/build_library.ts

```diff
@@ -277,6 +282,11 @@ export function mergeGlobal({
     );
   }
 
+  // We need to copy over any interfaces
+  for (const interfaceDeclaration of globalInterfaces) {
+    addInterfaceDeclaration(targetSourceFile, interfaceDeclaration);
+  }
```

Comment:
How long does running this script take? There's a few easy wins that could be implemented to speed up the performance if you find it's slow.

For example, bulking these all together so instead of doing...

```ts
for (const interfaceDeclaration of globalInterfaces)
    targetSourceFile.addInterface(interfaceDeclaration.getStructure());
```

...the following could be done:

```ts
targetSourceFile.addInterfaces(globalInterfaces.map(i => i.getStructure()));
```

That will be way faster because it will insert everything at the same time. Each manipulation to the source file requires reparsing and mapping the old nodes to the new nodes so doing that many times adds up (I have slowly been implementing performance improvements internally and hope to one day have incremental parsing).

I'll need to update the documentation with that recommendation. Anyway, I happened to be popping by the repo and saw this PR so I thought I'd take a look. Looks good! :)

Reply:
Right now it's just one interface so it's rather fast 😅 I was aiming for MVP with this PR hence `addInterface`. Thanks for explaining the internals - I had my eye on `addInterfaces` and now I'll definitely update the code to use it!